pkg/compare/parsing.go

```diff
@@ -179,21 +220,62 @@ func parseDiffConfig(filePath string) (UserConfig, error) {
 
 const noValue = "<no value>"
 
-func executeYAMLTemplate(temp *template.Template, params map[string]any) (*unstructured.Unstructured, error) {
+func executeYAMLTemplatRaw(temp *template.Template, params map[string]any) ([]byte, error) {
 	var buf bytes.Buffer
 	err := temp.Execute(&buf, params)
 	if err != nil {
-		return nil, fmt.Errorf("failed to constuct template: %w", err)
+		return []byte{}, fmt.Errorf("failed to constuct template: %w", err)
+	}
+	return buf.Bytes(), nil
+}
+func executeYAMLTemplate(temp *template.Template, params map[string]any) (*unstructured.Unstructured, error) {
+	content, err := executeYAMLTemplatRaw(temp, params)
+	if err != nil {
+		return nil, err
 	}
 	data := make(map[string]any)
-	err = yaml.Unmarshal(bytes.ReplaceAll(buf.Bytes(), []byte(noValue), []byte("")), &data)
+	err = yaml.Unmarshal(bytes.ReplaceAll(content, []byte(noValue), []byte("")), &data)
 	if err != nil {
-		return nil, fmt.Errorf("template: %s isn't a yaml file after injection. yaml unmarshal error: %w. The Template After Execution: %s", temp.Name(), err, buf.String())
+		return nil, fmt.Errorf("template: %s isn't a yaml file after injection. yaml unmarshal error: %w. The Template After Execution: %s", temp.Name(), err, string(content))
 	}
 	return &unstructured.Unstructured{Object: data}, nil
 }
-
-func extractMetadata(t *template.Template) (*unstructured.Unstructured, error) {
-	yamlTemplate, err := executeYAMLTemplate(t, map[string]any{})
+func extractMetadata(t *ReferenceTemplate) (*unstructured.Unstructured, error) {
+	yamlTemplate, err := t.Exec(map[string]any{})
 	return yamlTemplate, err
 }
+
+func extractCommentMap(temp *template.Template) (map[string]string, error) {
+	config := make(map[string]string)
+	raw, _ := executeYAMLTemplatRaw(temp, map[string]any{})
+	scanner := bufio.NewScanner(bytes.NewReader(raw))
+	errs := make([]error, 0)
+	for scanner.Scan() {
```

Comment:
In general we should try to make this function as elegant as possible, see my comment about switching to yaml,

that way this part should only extract the yaml from the comments and load it in to a struct that, 
will automatically make sure that there are no other unsupported keys in the config.